docs/ebpf/concepts/elf-sections.md

```diff
@@ -0,0 +1,207 @@
+Executable Linkable Format (ELF) is the standard application binary format for
```

Comment:
> Is starting at ELF useful here? By putting it in the title we assume that the reader knows that LLVM emits ELF, and that the library consumes them.

I think that's okay. What are the alternatives for making the user form a mental model? If not ELF sections, what kind of sections are they?

We could always make this first paragraph folded by default so the curious reader can explore if they want to. Agree on focusing on `SEC` instead of giving the full background first, but I think this primer needs to be given at some point along the way.

Reply:
I'm saying: don't begin this from the ELF side. If you want to do a paragraph on it later on that is fine.

Reply:
Agreed, I rewrote the first sections to omit the ELF term and only introduce it later. I've hidden the exploration steps behind a foldable admonition and extended it a bit to include llvm-objdump and bpftool.